src/authHooks.js

```diff
@@ -0,0 +1,75 @@
+import axios from 'axios';
+import { useEffect, useReducer } from 'react';
+import { useCookies } from 'react-cookie';
+import { authFieldReducer, useAxiosConfig } from './authMixin';
+
+const initialSignupFormValue = {
+  email: '',
+  password: '',
+  name: '',
+};
+
+const initialSigninValue = {
+  email: '',
+  password: '',
+};
+
+const useSignup = () => {
+  const [signupValue, dispatch] = useReducer(
+    authFieldReducer,
+    initialSignupFormValue
+  );
+
+  const [axiosConfig, axiosDispatch] = useAxiosConfig();
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    axios(axiosConfig)
+      .then((res) => {
+        alert(res.data);
+        window.location.replace('/vote');
+      })
+      .catch((err) => {
+        alert(err.response.data);
+        dispatch({ type: '' });
+      });
+  };
+
+  useEffect(() => {
+    axiosDispatch({ type: 'signup', data: signupValue });
+  }, [signupValue]);
+
+  return [signupValue, dispatch, handleSubmit];
+};
+
+const useSignin = () => {
+  const [signinValue, dispatch] = useReducer(
+    authFieldReducer,
+    initialSigninValue
+  );
+
+  const [cookie, setCookie] = useCookies(['JWTToken']);
```

Comment:
```suggestion
  const [, setCookie] = useCookies(['JWTToken']);
```
Destructuring 사용시 이렇게 하면 불필요한 변수 선언을 막을 수 있습니다